Initial DCS ATP setup

martin-bucinskas opened the pull request:
This is currently a spring service, but can be rewritten to something else. All it's relying upon is reactive web client along with some autowiring  for signing and encrypting.

## Whats changed
- Adding the gradle wrappers for the DCS ATP service.
- The service takes a JSON payload, signs, encrypts and signs again before submitting to DCS (currently mocked).
- Keys are currently packaged inside the built JAR until PKI is sorted and actual deployment pipelines are sorted out.
- Adding manifest for PaaS (should be moved out to the config repo).

martin-bucinskas committed:
dcs-setup: Adding gradle wrappers.

- Adding the gradle wrappers for the DCS ATP service.

martin-bucinskas committed:
dcs-setup: Adding the DCS service.

- The service takes a JSON payload, signs, encrypts and signs again before submitting to DCS (currently mocked).
- Keys are currently packaged inside the built JAR until PKI is sorted and actual deployment pipelines are sorted out.

martin-bucinskas committed:
dcs-setup: Adding manifest and cfignore for PaaS.

martin-bucinskas committed:
dcs-setup: Updating readme.

martin-bucinskas committed:
dcs-setup: Updating readme.

martin-bucinskas committed:
dcs-setup: Adding dockerfile for local env.

martin-bucinskas committed:
dcs-setup: Updating the cert gen to not present dialog.

kerrr commented on src/main/java/uk/gov/cabinetoffice/di/ipv/dcsatpservice/DcsAtpServiceApplication.java:
```diff
@@ -0,0 +1,13 @@
+package uk.gov.cabinetoffice.di.ipv.dcsatpservice;
```
I think we need to have `cabinetoffice` in the package, a root of `uk.gov.di.ipv` should be fine for our stuff.

martin-bucinskas replied:
Ok, will change it out to a shorter package name

kerrr reviewed (dismissed):
Looks good 👍🏼
Let's shorten the package name.
I'm not familiar with the reactor stuff, so would be good to talk about it.

martin-bucinskas committed:
dcs-setup: Updating dockerfile and package names.

- Renamed package from uk.gov.cabinetoffice.di to uk.gov.di.
- Only copying a single jar over for docker.

rcjenkins reviewed (dismissed):
LGTM

martin-bucinskas committed:
dcs-setup: Updating dockerfile.
